platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/FileSigningUtils.java

```diff
@@ -0,0 +1,402 @@
+
+    /**
+     * Generates a signature file for the given stream file
+     *
+     * @param destinationDirectory the directory to which the signature file will be saved
+     * @param streamType           type of the stream file
+     * @param streamFileToSign     the stream file to be signed
+     * @param keyPair              the keyPair used for signing
+     */
+    public static void signStreamFile(
+            @NonNull final File destinationDirectory,
+            @NonNull final StreamType streamType,
+            @NonNull final File streamFileToSign,
+            @NonNull final KeyPair keyPair) {
+
+        Objects.requireNonNull(destinationDirectory, "destinationDirectory must not be null");
+        Objects.requireNonNull(streamType, "streamType must not be null");
+        Objects.requireNonNull(streamFileToSign, "streamFileToSign must not be null");
+        Objects.requireNonNull(keyPair, "keyPair must not be null");
+
+        final String signatureFilePath = buildSignatureFilePath(destinationDirectory, streamFileToSign);
+
+        if (!streamType.isStreamFile(streamFileToSign)) {
+            System.err.println("File " + streamFileToSign + " is not a " + streamType);
+            return;
+        }
+
+        try {
+            final int version = readFirstIntFromFile(streamFileToSign);
+            if (version != SUPPORTED_STREAM_FILE_VERSION) {
+                System.err.printf(
+                        "Failed to sign file [%s] with unsupported version [%s]%n",
+                        streamFileToSign.getName(), version);
+                return;
+            }
+
+            final Hash entireHash = computeEntireHash(streamFileToSign);
+            final com.swirlds.common.crypto.Signature entireHashSignature = new com.swirlds.common.crypto.Signature(
+                    SignatureType.RSA, signData(entireHash.getValue(), keyPair));
+
+            final Hash metaHash = computeMetaHash(streamFileToSign, streamType);
+            final com.swirlds.common.crypto.Signature metaHashSignature =
+                    new com.swirlds.common.crypto.Signature(SignatureType.RSA, signData(metaHash.getValue(), keyPair));
+
+            writeSignatureFile(
+                    entireHash, entireHashSignature, metaHash, metaHashSignature, signatureFilePath, streamType);
+
+            System.out.println("Generated signature file: " + signatureFilePath);
+        } catch (final SignatureException | InvalidStreamFileException | IOException e) {
+            System.err.println("Failed to sign file " + streamFileToSign.getName() + ". Exception: " + e);
+        } catch (final InvalidKeyException | NoSuchAlgorithmException | NoSuchProviderException e) {
+            throw new RuntimeException("Irrecoverable error encountered", e);
+        }
+    }
+
+    /**
+     * Generates a signature file for the given file
+     * <p>
+     * File types known to be signed via this method are: event stream v3, record stream v2, and account balance files.
```

Comment:
This class should only sign event stream files. As such, method (and perhaps other names) can be more specific and informative.